### DISCUSSION TITLE
Is it just me, or is ghci+ncurses broken? (arm_big_sur)

### DISCUSSION THREAD

**Discussion by aryairani** [Homebrew/discussions#1649] [Everyday usage]
The ghci console seems broken for me.  Steps to reproduce:
```
$ brew install ghc  
==> Pouring ghc--8.10.5.arm64_big_sur.bottle.tar.gz
$ ghci
<type the following 9 keystrokes>:    a s d f ctrl-a . . . .
```
expected result: `Prelude> ....asdf`
actual result: `Prelude> asd.asd.asd.asd.asdf`

**Comment by SMillerDev** [comment]
Are you sure this is a homebrew packaging issue? Does the same happen with the x86 version? Does the same happen when you compile from source?

**Reply by aryairani** [reply]
I will try to get these answers.

**Reply by aryairani** [reply]
* bottled 8.10.5 on arm: above problem.
* bottled 8.10.5 on x86: seems fine.  
* building 8.10.5 on arm: same problem.
* building 8.10.5 on x86: seems fine.
* building 8.10.4 on arm
  * error: /bin/sh ./config.sub arm_vortex_tempest-apple-darwin20 failed
  * Do not report this issue to Homebrew/brew or Homebrew/core! 😅

**Reply by SMillerDev** [reply]
Did you compile that using `brew install --build-from-source`? Or did you build it outside of brew? Because if it's the latter that seems like an upstream bug, not brew.

**Reply by aryairani** [reply]
Yes, all of the above is with `brew install` or `brew install -s`.

**Comment by sourtin** [comment]
I'm seeing the same issue, all good on x86 but broken ncurses/readline on arm (both bottled and from source). I think this issue https://github.com/judah/haskeline/issues/162 is meant to solve it, but that doesn't help with the current homebrew version (8.10.7). Is there anything that can be done to bump versions here (presumably of the haskeline dependency), if that would help?

**Comment by sourtin** [comment]
For now, a fix seems to be to run ghci with `TERM=linux`. I'm currently just aliasing ghci and cabal to `TERM=linux ghci` etc.
